cpp/tests/utilities/table_utilities.cu

```diff
@@ -19,5 +19,15 @@ void expect_tables_equal(cudf::table_view lhs, cudf::table_view rhs) {
   }
 }
 
+/**
+ * @copydoc cudf::test::expect_tables_equivalent
+ * 
+ **/
+void expect_tables_equivalent(cudf::table_view lhs, cudf::table_view rhs) {  
+  for (auto i=0; i<lhs.num_columns(); ++i) {
```

Comment:
`lhs.num_columns()` may be called repeatedly in this loop (unless there is a guarantee the compiler can cache the `num_calls()` call, in which case please ignore this comment). You might consider one of the following:

1. store the `num_columns()` result before using it;
2. use (const) iterators; 
3. if possible, use range-based loops (probably not possible, without additional expensive machinery);
4. use `std::for_each()`

Reply:
Updated.  Note :  there's no for_each() that takes two iterator ranges, unfortunately.

Reply:
`std::transform()` does (...take 2 ranges)